homeassistant/components/bluetooth/__init__.py

```diff
@@ -178,14 +181,83 @@ def async_track_unavailable(
     return manager.async_track_unavailable(callback, address)
 
 
+async def _async_has_bluetooth_adapter() -> bool:
+    """Return if the device has a bluetooth adapter."""
+    if platform.system() == "Darwin":  # CoreBluetooth is built in on MacOS hardware
+        return True
+    if platform.system() == "Windows":  # We don't have a good way to detect on windows
+        return False
+    return bool(await _async_get_bluetooth_adapters())
+
+
+async def _async_get_bluetooth_adapters() -> set[str]:
+    """Return a list of bluetooth adapters."""
+    adapters: set[str] = set()
+    try:
+        bus = await MessageBus(
+            bus_type=BusType.SYSTEM, negotiate_unix_fd=True
+        ).connect()
+    except FileNotFoundError:
+        return adapters
+    msg = Message(
+        destination="org.bluez",
+        path="/",
+        interface="org.freedesktop.DBus.ObjectManager",
+        member="GetManagedObjects",
+    )
+    if (
+        not (reply := await bus.call(msg))
+        or reply.message_type != MessageType.METHOD_RETURN
+    ):
+        return adapters
+    for path in reply.body[0]:
+        path_str = str(path)
+        if path_str.startswith("/org/bluez/hci"):
+            split_path = path_str.split("/")
+            adapters.add(split_path[3])
+    return adapters
+
+
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the bluetooth integration."""
+    if hass.config_entries.async_entries(DOMAIN):
+        return True
+    if DOMAIN in config:
```

Comment:
Should only auto import if it is explicitly listed in `configuration.yaml`.

`default_config` should only trigger the discovery path if its available.